### PR TITLE
Mcj 06

### PR DESCRIPTION
**Pull request by ThomassenCathleia**
The basic model structure of this project

### CODE REVIEW

File: src/be/mcjava/model/Ingredient.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    public int getCurrentStock() {
+        return currentStock;
+    }
+    
+    public void setCurrentStock(int currentStock) {
+        this.currentStock = currentStock;
+    }
+    
+    public boolean isAvailable() {
+        //todo: implement this, returns false so code will compile -Michiel 24/5/2019
```
Comment (lenshuygh) [inline src/be/mcjava/model/Ingredient.java]: isn't this already implementable now?
i would return (currentStock > 0); here

File: src/be/mcjava/model/CustomerOrder.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    public List<OrderItem> getOrderedItems() {
+        return orderedItems;
+    }
+    
+    public void setOrderedItems(List<OrderItem> orderedItems) {
+        this.orderedItems = orderedItems;
+    }
+    
+    public void addToOrder(OrderItem orderItem) {
+    }
```
Comment (lenshuygh) [inline src/be/mcjava/model/CustomerOrder.java:77]: implementation can be done now with:
this.orderedItems.add(orderItem);
?

File: src/be/mcjava/model/CustomerOrder.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    
+    public void removeFromOrder(OrderItem orderItem) {
+    }
```
Comment (lenshuygh) [inline src/be/mcjava/model/CustomerOrder.java:81]: implement;
this.orderedItems.remove(orderItem);
?

File: src/be/mcjava/model/Product.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    public BigDecimal getPrice() {
+        return price;
+    }
+    
+    public void setPrice(BigDecimal price) {
+        this.price = price;
+    }
+    
+    public boolean isAvailable() {
+        //todo: implement this, returns false so code will compile -Michiel 24/5/2019
```
Comment (lenshuygh) [inline src/be/mcjava/model/Product.java]: implementation?
for (Ingredient ingredient : ingredients.keySet()) {
            if(ingredient.getCurrentStock() < ingredients.get(ingredient)){
                return false;
            }
        }
        return true;

**Comment by lenshuygh** [comment]
Great work Michiel, thats a lot of work done!
Added a few comments on the helper-methods that can already be implemented...

**Comment by ThomassenCathleia** [comment]
The main reason why I did not implement anything seriously was to just lay down the structure.

The boolean return methods are rather simple. The add/remove methods might need some better implementation, such as needing a custom equals and compare methods.

**Comment by lenshuygh** [comment]
I get what you mean, but now they'll need to be implemented by making another story , if the methods are available and they don't do anything it will create confusion, especially since some do return something thats not a real result (eg: return false;)
If the method is available to other code you can't expect the person working on another story to go and check all your methods...
You can merge it as is but you'll need to add another story to make the implementations

**Review by lenshuygh — APPROVED** [review]
good stuff :-)

**Review by Fonthom — APPROVED** [review]

**Pull request merged**